File: src/main/scala/Point.scala
trait Debug {
  def debugVars(): Unit = {
    val fields = this.getClass.getDeclaredFields
    fields.foreach{ field =>
      field.setAccessible(true)
      println("Field: " + field.getName + " => " + field.getType + ", " + field.get(this))}
  }
}

class Point(xv: Int, yv: Int) extends Debug {
  var x: Int = xv
  var y: Int = yv
  var a: String = "test"
}

object Point {
  def main(args: Array[String]): Unit = {
    val point = new Point(3, 4)
    point.debugVars()
  }
}

File: src/main/java/Lab8/Point.java
package Lab8;

class Point {
    private int x, y;
    private String a, b;

    public Point(int x, int y) {
        this.x = x;
        this.y = y;
        this.a = "test";
        this.b = "another";
    }

    public static void main(String[] args)  {
        Point point = new Point(3, 4);
        Debug.fields(point);
    }
}
